### PR TITLE
[Feature] Agency Cache

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

**Review by jwierzbo — APPROVED** [review]

File: pkg/deployment/agency/state.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	state := c[0].Arango
+
+	if _, ok := state.Current.Collections["_system"]; !ok {
+		return State{}, errors.Newf("Unable to find system database (invalid data)")
+	}
+
+	if _, ok := state.Plan.Collections["_system"]; !ok {
+		return State{}, errors.Newf("Unable to find system database (invalid data)")
```
Comment (informalict) [inline pkg/deployment/agency/state.go:74]: Unable to find system database (invalid data) -> Unable to find system database in the plan (invalid data)

File: pkg/deployment/agency/state.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
+//
+// DISCLAIMER
+//
+// Copyright 2016-2021 ArangoDB GmbH, Cologne, Germany
```
Comment (informalict) [inline pkg/deployment/agency/state.go:4]: why 2016. it is a new file and should start with 2021

**Review by informalict — APPROVED** [review]
LGTM with two comments

File: pkg/deployment/agency/state.go (hunk 2 above)
Comment (ajanikow) [reply pkg/deployment/agency/state.go:4]: We use everywhere licenses like this

File: pkg/deployment/agency/state.go (hunk 1 above)
Comment (ajanikow) [reply pkg/deployment/agency/state.go:74]: This error should never happen, unless agency is not totally broken

**Pull request merged**